crucible-llvm/src/Lang/Crucible/LLVM/SimpleLoopInvariant.hs

```diff
@@ -0,0 +1,1151 @@
+        return left
+      else case MapF.lookup (C.llvmPointerOffset (C.regValue left)) (varSubst subst) of
+        Just join_entry -> do
+          -- liftIO $ ?logMessage $
+          --   "SimpleLoopInvariant.joinRegEntry: LLVMPointerRepr: Just: "
+          --   ++ show (W4.printSymExpr $ bodyValue join_entry)
+          --   ++ " -> "
+          --   ++ show (W4.printSymExpr $ C.llvmPointerOffset (C.regValue right))
+          put $ subst{ varSubst =
+            MapF.insert
+              (C.llvmPointerOffset (C.regValue left))
+              (join_entry { bodyValue = C.llvmPointerOffset (C.regValue right) })
+              (varSubst subst) }
+          return left
+        Nothing -> do
+          liftIO $ ?logMessage "SimpleLoopInvariant.joinRegEntry: LLVMPointerRepr: Nothing"
+          join_varaible <- liftIO $ W4.freshConstant sym
+                             (W4.safeSymbol "reg_join_var") (W4.BaseBVRepr w)
+          let join_entry = InvariantEntry
+                { headerValue = C.llvmPointerOffset (C.regValue left)
+                , bodyValue = C.llvmPointerOffset (C.regValue right)
+                }
+          put $ subst{ varSubst = MapF.insert join_varaible join_entry (varSubst subst) }
+          return $ C.RegEntry (C.LLVMPointerRepr w) $ C.LLVMPointer (C.llvmPointerBlock (C.regValue left)) join_varaible
+
+    | otherwise -> do
+      liftIO $ ?logMessage "SimpleLoopInvariant.joinRegEntry: LLVMPointerRepr, unequal blocks!"
+      havoc_blk <- ask
+      case MapF.lookup (C.llvmPointerOffset (C.regValue left)) (varSubst subst) of
+        Just _ -> do
+          -- widening varible already present in the var substitition.
+          -- we need to remove it, and add it to the havoc map instead
+          put subst { varSubst = MapF.delete (C.llvmPointerOffset (C.regValue left)) (varSubst subst)
+                    , varHavoc = MapF.insert (C.llvmPointerOffset (C.regValue left)) (Const ()) (varHavoc subst)
+                    }
+          return $ C.RegEntry (C.LLVMPointerRepr w) $ C.LLVMPointer havoc_blk (C.llvmPointerOffset (C.regValue left))
+
+        Nothing -> do
+          havoc_var <- liftIO $ W4.freshConstant sym (W4.safeSymbol "havoc_var") (W4.BaseBVRepr w)
+          put subst{ varHavoc = MapF.insert havoc_var (Const ()) (varHavoc subst) }
+          return $ C.RegEntry (C.LLVMPointerRepr w) $ C.LLVMPointer havoc_blk havoc_var
+
+  C.BoolRepr
+    | List.isPrefixOf "cmacaw" (show $ W4.printSymExpr $ C.regValue left) -> do
+      liftIO $ ?logMessage "SimpleLoopInvariant.joinRegEntry: cmacaw_reg"
+      return left
+    | otherwise -> do
+      -- liftIO $ ?logMessage $
+      --   "SimpleLoopInvariant.joinRegEntry: BoolRepr:"
+      --   ++ show (W4.printSymExpr $ C.regValue left)
+      --   ++ " \\/ "
+      --   ++ show (W4.printSymExpr $ C.regValue right)
+      join_varaible <- liftIO $ W4.freshConstant sym (W4.safeSymbol "macaw_reg") W4.BaseBoolRepr
```

Comment:
I'm a bit confused by this code. Why would we always name this `macaw_reg`? As far as I can tell, it's possible to hit this case even when neither `left` nor `right` have `macaw`-derived names.